### PR TITLE
adaptors_task SChyruk

### PR DESCRIPTION
**Pull request by ChNokia**

### CODE REVIEW

File: adaptors_task/student/src/impl.cpp [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+bool isValid(const std::string& source) {
+    std::stack<char> bracketsStack;
+
+    for (const char &ch : source) {
+        auto resultIter = std::find_if(brackets.begin(), brackets.end(), [ch](const std::pair<char, char> value) {
+            return value.first == ch;
+        });
+
+        if (resultIter != brackets.end()) {
+            bracketsStack.push((*resultIter).second);
```
Comment (smay1613) [inline adaptors_task/student/src/impl.cpp:28]: prefer "->"

File: adaptors_task/student/src/impl.cpp [hunk 2]
```diff
@@ -1,8 +1,40 @@
...
+        } else if (isCloseBracket(ch)) {
+            if (bracketsStack.empty()) {
```
Comment (smay1613) [inline adaptors_task/student/src/impl.cpp:30]: you should check stack top value here too

File: adaptors_task/student/src/impl.cpp [hunk 3]
```diff
@@ -1,8 +1,40 @@
...
+    auto resultIter = std::find_if(brackets.begin(), brackets.end(), [bracket](const std::pair<char, char> value) {
+        return value.second == bracket;
+    });
+    return resultIter != brackets.end();
+}
+
+bool isValid(const std::string& source) {
+    std::stack<char> bracketsStack;
+
+    for (const char &ch : source) {
```
[the remark this answers is not in the document]
Comment (ChNokia) [reply adaptors_task/student/src/impl.cpp:22]: done

File: adaptors_task/student/src/impl.cpp (hunk 1 above)
Comment (ChNokia) [reply adaptors_task/student/src/impl.cpp:28]: done

File: adaptors_task/student/src/impl.cpp (hunk 2 above)
Comment (ChNokia) [reply adaptors_task/student/src/impl.cpp:30]: I don't understand what do you mean.

Comment (smay1613) [reply adaptors_task/student/src/impl.cpp:30]: I mean the case when stack top value (open bracket) has another type

Comment (ChNokia) [reply adaptors_task/student/src/impl.cpp:30]: "another type" is it mean other symbol? if other symbol I skip it and add only brackets to the stack or I something don't understand

Comment (smay1613) [reply adaptors_task/student/src/impl.cpp:30]: See this example:
[{]}
Stack:
[
{
Then, next symbol is "]" - so, bracket on top of the stack ("}") is not the same type. You can return immediately.

Comment (ChNokia) [reply adaptors_task/student/src/impl.cpp:30]: done. thanks

File: adaptors_task/student/src/impl.cpp [hunk 4]
```diff
@@ -1,8 +1,40 @@
...
+            return value.first == symbol;
+        });
+
+        if (resultIter != brackets.end()) {
+            bracketsStack.push(resultIter->second);
+        } else if (isCloseBracket(symbol)) {
+            if (bracketsStack.empty() || bracketsStack.top() != symbol) {
+                return false;
+            }
+            if (bracketsStack.top() == symbol) {
```
Comment (smay1613) [inline adaptors_task/student/src/impl.cpp:33]: you can remove this check

Comment (ChNokia) [reply adaptors_task/student/src/impl.cpp]: done

**Pull request closed without merging**